p2p/host/basic/basic_host.go

```diff
@@ -157,14 +161,29 @@ func NewHost(ctx context.Context, net network.Network, opts *HostOpts) (*BasicHo
 	}
 
 	var err error
+
+	localIpv4, err := getLocalIPAddrFor(net.IPv4(0, 0, 0, 0))
+	if err != nil {
+		log.Debugw("failed to fetch local IPv4 address", "err", err)
+	} else {
+		h.localIPv4Addr = localIpv4
```

Comment:
This can change at runtime. We can cache it, but we need to re-evaluate it semi frequently.

Reply:
Can you explain how/when it would change at runtime ? Would it happen when the Router assigns a new private IP to the machine running this libp2p node ?

Reply:
Yes. E.g., we change networks.